### PR TITLE
[SYSTEMML-1608] adding ALS demo notebook

### PR DESCRIPTION
**Pull request by iyounus**
This PR adds ALS example running in python notebook.

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1473/

**Comment by iyounus** [comment]
@dusenberrymw could you please have a look at this. Let me know if this needs and addition/subtraction.

File: samples/jupyter-notebooks/ALS_python_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+    "# inject 0s in U if original R had empty rows\n",
+    "if (num_zero_rows > 0) {\n",
+    "  U = removeEmpty (target = diag (orig_nonzero_rows_ind), margin = \"cols\") %*% U;\n",
+    "}\n",
+    "# inject 0s in R if original V had empty rows\n",
+    "if (num_zero_cols > 0) {\n",
+    "  M = removeEmpty (target = diag (orig_nonzero_cols_ind), margin = \"cols\") %*% M;\n",
+    "}\n",
+    "M = t(M);\n",
+    "\"\"\""
```
Comment (dusenberrymw) [inline samples/jupyter-notebooks/ALS_python_demo.ipynb]: Can you remove the extra spaces between the function names and the opening parentheses throughout this script?  That issue is fairly prevalent in the older scripts, and it's caused confusion due to being hard to read.

I.e. `rand (rows = m, cols = r, min = -0.5, max = 0.5)` --> `rand(rows = m, cols = r, min = -0.5, max = 0.5)`

File: samples/jupyter-notebooks/ALS_python_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+    "\n",
+    "lambda_I = diag (matrix (lambda, rows = r, cols = 1));\n",
+    "it = 0;\n",
+    "converged = FALSE;\n",
+    "while ((it < max_iter) & (!converged)) {\n",
+    "  it = it + 1;\n",
+    "  # keep M fixed and update U\n",
+    "  parfor (i in 1:m) {\n",
+    "    M_nonzero_ind = t(R[i,] != 0);\n",
+    "    M_nonzero = removeEmpty (target=M * M_nonzero_ind, margin=\"rows\");\n",
```
Comment (dusenberrymw) [inline samples/jupyter-notebooks/ALS_python_demo.ipynb]: A comment here might help to explain that this is temporarily removing associated feature rows for any missing ratings.

File: samples/jupyter-notebooks/ALS_python_demo.ipynb [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running the Algorithm\n",
+    "\n",
+    "We'll first create MLContext object which the entry point for SystemML. Inputs and outputs are defined through dml function."
```
Comment (dusenberrymw) [inline samples/jupyter-notebooks/ALS_python_demo.ipynb]: nitpick: "create **an** MLContext ..."
nitpick: "though **a** dml function."

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1476/

**Comment by deroneriksson** [comment]
@dusenberrymw Can we merge this?

**Comment by iyounus** [comment]
@deroneriksson @dusenberrymw Give me one more day on this. I want to add a plot for loss function.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1479/

**Comment by deroneriksson** [comment]
LGTM. Thank you @iyounus! I'll merge

**Pull request closed without merging**